### ISSUE TITLE
[Intern Progress] Ademola Omosanya - Erisilia Contribution Outreachy

### ISSUE THREAD

**Issue by ademolaomosanya** [ersilia-os/outreachy-contributions#26]
# 🌟 Intern Progress Tracker  
Welcome! Use this issue to track your progress as you work through the AI/ML modeling exercise. Follow each step, check the boxes when completed, and update your issue as you go.

# ▶️ **Entry tasks**
- [x] Motivation letter posted in this issue
- [x] Installed the Ersilia Model Hub and successfully run models!
- [x] Recorded system settigns in this issue
- [x] Explained completed tests of Ersilia in the issue

---

##  **1. Download a Dataset of Interest**
- [x] Created a fork of this repository to work on
- [x] Selected a dataset from Therapeutics Data Commons (TDC)
- [x] Verified that the dataset is a classification problem*
- [x] Read and understood the dataset’s background and endpoint  
- [x] Assessed computational feasibility (size, features, etc.)  
- [x] Downloaded the dataset using Python  
- [x] Saved data in the `/data` folder  
- [x] Started writing documentation in the README (Installation, dataset details, how to run code)  

---

##  **2. Featurise the Data**
- [x] Browsed Ersilia Model Hub for "Representation" models  
- [x] Selected an appropriate featuriser and justified my choice  
- [x] Implemented featurisation in a notebook or script
- [x] Ensured the process is reproducible 
- [x] Updated the README with featurisation instructions  

---

##  **3. Build an ML Model**
- [x] Selected an ML framework
- [x] Trained the model on featurised data  
- [x] Evaluated the model (accuracy, precision, recall, etc.)  
- [x] Created graphs using Matplotlib for better visualization  
- [x] Ensured all steps are reproducible  
- [x] Documented the model-building process in README

---

##  **4. Prepare Code for Review**
- [x] Double-checked that documentation is clear and complete  
- [x] Evaluated if my model performance is satisfactory 
- [x] Identified possible improvements  
- [x] Updated this issue to request a review 

---

##  **5. Stretch Tasks (Optional)**
- [ ] Tried a different **featuriser**  
- [ ] Tested other **ML architectures**  
- [ ] Applied the model to **public datasets (e.g., ChEMBL)**  
- [ ] Any other improvements? (Describe below)  

---

## 📌 **Requesting Review**
Once you’ve completed **Steps 1-4**, request a review by mentioning a mentor:  
👤 *@GemmaTuron* Please review my work!  

---

### 📝 **Asking questions**
Use this issue to update your progress, explain where you are stuck or ask for review!

Thank you for contributing! 🎉

**Comment by ademolaomosanya** [comment]
Hello @GemmaTuron 

## Motivation letter

 **Enhancing the Ersilia Model Hub, an AI platform for drug discovery**

My name is Ademola Omosanya, i am a self-taught programmer with  experience in full-stack development.
i am passionate about tech and that is what  drives me to keep pushing in my tech career. i have background in software engineering with enthausiam in Al/ML so i enrolled in  an AI/ML course at udemy that covered key topics such as data preprocessing, supervised and unsupervised learning, model evaluation, and deployment. 

**My techical expertise : -**

My background is with web development before going into Data science, my tech stacks includes Python, JavaScript, Node.js, React, Next.js, and MySQL, along with experience working with Supabase and Conda a tool widely used in data science and machine learning.
Additionally, my experience with Swagger API documentation will be valuable in improving the accessibility of AI models for researchers in the field of drug discovery.
i have also contributed to an open source project before at Ocaml during a previous outreachy contribution phase. this helped me collaborate within the open-source community a skill set directly applicable to this project

**My Goals and Passion for Erisillia : -**

My goal is to get into the AI/ML tech space as an open source contributor and make meaningful impacts through domains like climate science, drug discovery, and financial technology. By participating in this project through Erisillia will allow me to enhance my skills in AI model integration while contributing to a platform that advocate drug discovery for researchers worldwide.

Thank you!

**Comment by ademolaomosanya** [comment]
## Installation of  the Ersilia Model Hub and successfully run models

**I have  installed the Ersilia Model Hub and ran the models.**
i used the following commands for the models:
- ersilia -v fetch eos3b5e
- ersilia serve eos3b5e
- ersilia -v run -i "CCCC"

I fetched eos3b5e using the command : ersilia -v fetch eos3b5e

<img width="1110" alt="Image" src="https://github.com/user-attachments/assets/8ee16315-3e02-49b2-96ba-044d7b39d3e5" />



- **eos3b5e** 

<img width="1140" alt="Image" src="https://github.com/user-attachments/assets/a8e0b167-f3e0-42e0-88b8-3c2270129ef6" />


 - Results

<img width="1247" alt="Image" src="https://github.com/user-attachments/assets/f37ebda1-63d6-4ab4-bc1d-06388b4ad887" />

**Comment by ademolaomosanya** [comment]
## My System Settings

- OS:  MacOS 12.7.6
- Python Version: Python 3.12.2
- Conda Version: conda 25.1.1
- Docker version 24.0.6

<img width="1001" alt="Image" src="https://github.com/user-attachments/assets/44c3b39e-84fc-49fd-a7e8-6fb9d7a6e70a" />

**Comment by ademolaomosanya** [comment]
👤 @GemmaTuron Please review my work!

Thank you.

**Comment by ademolaomosanya** [comment]
@GemmaTuron 
i updated my Motivation talk, the previous one was ChatGPT edited as i wasnt aware of the guideline before now.
Kindly review my work as i await feed back.
Thank you

**Comment by GemmaTuron** [comment]
Hello @ademolaomosanya 

Please remember to mark the tasks on the issue as this is what I'll look at first. Thanks for working on your motivation letter, please be careful with misspellings (Ersilia). I see you have mostly self taught yourself programming, great endeavour.
Please go ahead and submit the contribution via the outreachy website

**Comment by ademolaomosanya** [comment]
Thank you @GemmaTuron 
I have marked the tasks.
And recorded my contribution to outreachy 
Also I will like to add that being a self taught developer as being my biggest motivation because against all odds I tried to learn and make meaningful impact with my tech skills.
Looking forward to contributing to Erisilia

**Comment by GemmaTuron** [comment]
Hi @ademolaomosanya 

Are you still interested in contributing to this period? If so please get started on week 2 and 3 tasks.

**Comment by ademolaomosanya** [comment]
Hi @GemmaTuron ,

Yes, I’m still interested in contributing this period. I’ll get started on the Week 2 and 3 tasks right away. Thanks for the update!

**Comment by GemmaTuron** [comment]
Ok @ademolaomosanya 

A reminder that on week 4 we will no longer provide feedback on the coding tasks and just focus on final applications.

**Comment by ademolaomosanya** [comment]
I’ve completed Steps 1-4, 
👤 @GemmaTuron Please review my work!
Here is my PR : https://github.com/ersilia-os/outreachy-contributions/pull/47

**Comment by GemmaTuron** [comment]
Hi @ademolaomosanya 

In order to provide feedback please write a summary here of the tasks attempted and the results you obtained, thanks.

**Comment by ademolaomosanya** [comment]
#  Development and Evaluation of a Machine Learning Model for Human Intestinal Absorption Prediction

 Description of the Task
  - The main goal was to use the TDC HIA_Hou dataset to create a machine learning model that would predict the Human 
      Intestinal Absorption (HIA) of medicinal molecules.  The assignment was:

**The dataset :**
 -  578 compounds with binary absorption labels were prepared.
 - Distribution of classes: 13.49% not absorbed, 86.51% absorbed
 - SMILES strings are used to depict molecular structures.
 
**Feature Engineering:**
 -  Complete molecular feature implementation
 - MACCS keys, physicochemical descriptors, and generated Morgan fingerprints
 -  Centralized configuration ensured reproducibility.
 
**Model Development:** 
 - Hyperparameter-optimized XGBoost classifier
  - SMOTE was used to address the class imbalance.
  -  5-fold cross-validation was used for a thorough assessment.

# Results
   Accuracy:    94.83%
   Precision:   95.05%
   Recall:      98.97%
   F1 Score:    96.97%
   ROC AUC:     98.05%
   MCC:         0.89

The developed model demonstrates excellent performance in predicting human intestinal absorption of drug compounds, achieving over 94% accuracy across all metrics. This tool can significantly aid in early-stage drug development by providing reliable absorption predictions.

![Image](https://github.com/user-attachments/assets/77e38dd5-e992-407f-9291-0ce45103178f)

**Comment by GemmaTuron** [comment]
Thanks @ademolaomosanya 

Please go ahead and submit your final application if you haven't yet

**Comment by miquelduranfrigola** [comment]
Auto-closing this Outreachy intern-progress tracker as completed — the contribution round has ended. Reopen if follow-up is needed.
